### PR TITLE
[MIG] scripts migration for upgrade safe from v8.0-project_task_category

### PR DESCRIPTION
**Pull request by joycehuan**
project_task_category ,business_requirement_resource upgrade  migrate

### CODE REVIEW

**Review by lonelysun — CHANGES_REQUESTED** [review]
fix your PR name

**Comment by pedrobaeza** [comment]
And there's already other PR for the same in #384

**Comment by lonelysun** [comment]
`And there's already other PR for the same in #384`
@pedrobaeza  This version is more reasonable.
@victormartinelicocorp  Maybe you can close your PR #384 .

**Comment by martintorresvm** [comment]
@joycehuan You might need add external_dependency on the module regards openupgradelib
and modify .travis.yml with the pip install for it

**Comment by joycehuan** [comment]
@victormartinelicocorp

File: project_task_category/migrations/10.0.1.0.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_model_renames = [('project.category.main', 'project.category'), ]
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    cr = env.cr
+    try:
```
Comment (martintorresvm) [inline project_task_category/migrations/10.0.1.0.0/pre-migration.py]: Why should fail this?
we might prefer to check if table, column, etc.. exist
My point remove try..except unless we can't control what could happends

File: project_task_category/migrations/10.0.1.0.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            """ alter table business_requirement_resource_template
+                drop constraint if exists
+                business_requirement_resource_template_categ_id_fkey;""")
+        if openupgrade.table_exists(cr, 'project_category_main') and \
+                not openupgrade.table_exists(cr, 'project_category'):
+            openupgrade.rename_tables(cr,
+                                      [('project_category_main',
+                                        'project_category'), ])
+            openupgrade.rename_models(cr, _model_renames)
+        print "project_category /  business_requirement_resource upgrade done."
```
Comment (martintorresvm) [inline project_task_category/migrations/10.0.1.0.0/pre-migration.py]: better use openupgrade.logging()

**Review by martintorresvm — CHANGES_REQUESTED** [review]

**Comment by joycehuan** [comment]
@victormartinelicocorp

File: project_task_category/migrations/10.0.1.0.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+_model_renames = [('project.category.main', 'project.category'), ]
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    cr = env.cr
+    openupgrade.lift_constraints(cr, 'project_task', 'categ_id')
+    openupgrade.lift_constraints(cr,
+                                 'business_requirement_resource_template',
```
Comment (elicoidal) [inline project_task_category/migrations/10.0.1.0.0/pre-migration.py]: I think there is a mismatch here: this should be in the business requirement related module. not here.

File: project_task_category/migrations/10.0.1.0.0/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            drop constraint if exists
+            business_requirement_resource_template_categ_id_fkey;""")
+    if openupgrade.table_exists(cr, 'project_category_main') and \
+            not openupgrade.table_exists(cr, 'project_category'):
+        openupgrade.rename_tables(cr,
+                                  [('project_category_main',
+                                    'project_category'), ])
+        openupgrade.rename_models(cr, _model_renames)
+    openupgrade.logging(
+        "project_category /  business_requirement_resource upgrade done."
```
Comment (elicoidal) [inline project_task_category/migrations/10.0.1.0.0/pre-migration.py]: remove extra space

**Review by elicoidal — CHANGES_REQUESTED** [review]
This is not the place for this migration script.
Please close and push to BR related repo

**Comment by joycehuan** [comment]
done @victormartinelicocorp @elicoidal  couldn't rename table, copy data from old table to new table in post-migrate.py

**Review by elicoidal — APPROVED** [review]
LGTM
Although i am not sure this PR and migration script should be in this repo or in openupgrade repo
cc @pedrobaeza

**Review by lonelysun — APPROVED** [review]

**Comment by elicoidal** [comment]
@victormartinelicocorp can you review so that we can merge?
cc @pedrobaeza @dreispt

**Review by martintorresvm — APPROVED** [review]
LGTM

File: project_task_category/__manifest__.py [hunk 5]
```diff
@@ -8,6 +8,11 @@
     'version': '10.0.1.1.0',
     'author': 'Elico Corp, Odoo Community Association (OCA)',
     'license': 'AGPL-3',
+    'external_dependencies': {
```
Comment (pedrobaeza) [inline project_task_category/__manifest__.py]: This shouldn't be added. It's not required for installing the module. Only for upgrading it.

File: project_task_category/migrations/10.0.1.0.0/post-migration.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# © 2018 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    cr = env.cr
+    if openupgrade.table_exists(cr, 'project_category_main') and \
```
Comment (pedrobaeza) [inline project_task_category/migrations/10.0.1.0.0/post-migration.py]: Do the contrary check and return for saving one indentation level on the rest of the code.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: project_task_category/__manifest__.py (hunk 5 above)
Comment (joycehuan) [reply project_task_category/__manifest__.py]: This pull request code is for upgrade the module.

File: project_task_category/migrations/10.0.1.0.0/post-migration.py (hunk 6 above)
Comment (joycehuan) [reply project_task_category/migrations/10.0.1.0.0/post-migration.py]: done

File: project_task_category/__manifest__.py (hunk 5 above)
Comment (elicoidal) [reply project_task_category/__manifest__.py]: @joycehuan still not necessary. 
Normal usage does not require it: you have to protect the import [here](https://github.com/OCA/project/pull/386/files#diff-d158c8129a3a1d7d4b2d6fb3366cd10aR4) though

File: project_task_category/__manifest__.py [hunk 7]
```diff
@@ -8,6 +8,11 @@
...
+        'python': [
+            'openupgradelib',
+        ]
+    },
     'category': 'Project Management',
     'website': 'https://www.elico-corp.com',
```
Comment (elicoidal) [inline project_task_category/__manifest__.py]: change to https://github.com/OCA/project

File: project_task_category/__manifest__.py (hunk 5 above)
Comment (joycehuan) [reply project_task_category/__manifest__.py]: done

File: project_task_category/__manifest__.py (hunk 7 above)
Comment (joycehuan) [reply project_task_category/__manifest__.py]: done

File: project_task_category/migrations/10.0.1.0.0/post-migration.py (hunk 6 above)
Comment (pedrobaeza) [reply project_task_category/migrations/10.0.1.0.0/post-migration.py]: I still don't see it

File: .travis.yml [hunk 8]
```diff
@@ -38,6 +38,7 @@ install:
   - git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
   - travis_install_nightly
+  - pip install --ignore-installed git+https://github.com/OCA/openupgradelib.git@master
```
Comment (pedrobaeza) [inline .travis.yml]: This is not needed.

File: project_task_category/migrations/10.0.1.0.0/post-migration.py (hunk 6 above)
Comment (joycehuan) [reply project_task_category/migrations/10.0.1.0.0/post-migration.py]: Sorry. Incorrect comprehension.  Now you can have a look.

**Review by pedrobaeza — APPROVED** [review]
OK, too verbose IMO, but it's not so important

**Pull request merged**